scripts/core/editor3/components/annotations/AnnotationInput.jsx

```diff
@@ -183,13 +179,23 @@ AnnotationInputBody.propTypes = {
     language: PropTypes.string,
     spellcheckerEnabled: PropTypes.bool,
     highlightsManager: PropTypes.object.isRequired,
+    annotationTypes: PropTypes.array.isRequired,
 };
 
 const mapStateToProps = (state) => ({
     language: state.item.language,
     spellcheckerEnabled: state.spellcheckerEnabled,
 });
 
+const AnnotationInputBodyWithDependenciesLoaded = connectPromiseResults(() => ({
+    annotationTypes: new Promise((resolve) => {
+        ng.get('metadata').initialize()
```

Comment:
this should be a promise already, wouldn't this work?
```
annotationTypes: ng.get('metadata').initialize().then(() => ng.get('metadata').values.annotation_types)
```

Reply:
Yep, I guess it would. Will commit a fix